sound/soc/sof/imx/imx8.c

```diff
@@ -364,7 +364,11 @@ static int imx8_remove(struct snd_sof_dev *sdev)
 /* on i.MX8 there is 1 to 1 match between type and BAR idx */
 static int imx8_get_bar_index(struct snd_sof_dev *sdev, u32 type)
 {
-	return type;
+	/* Only IRAM and SRAM bars are valid */
+	if (type == SOF_FW_BLK_TYPE_IRAM || type == SOF_FW_BLK_TYPE_SRAM)
+		return type;
```

Comment:
a *soft* comment - maybe a `switch` would look better here and in imx8m too of course, but not a reason for a new version, only if you have to re-spin it for some other reason.

Reply:
@ujfalusi lets go with the switch as in the future we will also add DRAM bar. Other than that it looks good to me. 

We can merge this after the change.

Reply:
@lyakh, @dbaluta, good point, I'll update it and push a new version (after lunch)